app/server.R

```diff
@@ -1,125 +1,119 @@
-      smr_audit
-    }
-  })
   
-  #observe the audit selected by user and update the choices according to the audit selection
-  observeEvent(input$SMRaudit, {
-    updateSelectInput(session, inputId = "Year", choices = c(unique(audit()$year)))
+
+### SMR Audit ---------------------------------------------------------------
+
+
+  ##Filters for SMR and health board
+  filters1 <- reactive({
+    smr_audit %>%
+      filter(case_when(input$SMRaudit %in% unique(smr_audit$audit) ~ audit==input$SMRaudit,
+                       TRUE ~ audit == audit),
+             case_when(input$Healthboard %in% unique(smr_audit$healthboard) ~ healthboard == input$Healthboard,
+                       TRUE ~ healthboard == healthboard))
   })
   
+  ##Filters for the Year and Data Item Name, they both depend on the SMR and HB chosen by user
   
-  #reactive year selection, returns data filtered by the year selected 
-  year <- reactive({audit()%>%
-      filter(year == input$Year)})
-  
-  # year <- reactive({
-  #   if(input$Year %in% audit()$Year){
-  #     audit() %>%
-  #       filter(Year == input$Year)
-  #   }
-  #   else {
-  #     audit()
-  #   }
-  # })
-  
-  
-  #observe the year selected and update the health board choices, the (All) option is added so that users can view all HBs by default
-  observeEvent(input$Year, {
-    updateSelectInput(session, inputId = "Healthboard", choices = c("(All)", unique(year()$healthboard)))
+    #Since Data Item Name and SMR depend on multiple inputs, we store the inputs they rely on in a reactive list
+  to_listen_audit <- reactive({
+    list(input$SMRaudit, input$Healthboard)
   })
-  
-  #reactive health board selection, if a specific HB is selected filter the data further, else return the table reaturned by the reactive year()
-  healthboard <- reactive({
-    if(input$Healthboard %in% year()$healthboard){
-      year()%>%
-        filter(healthboard == input$Healthboard)
-    }
-    else {
-      year()
-    }
+    #Update Year selection based on inputs
+  observeEvent(to_listen_audit(), {
+    updateSelectInput(session, inputId = "Year", choices = c("(All)",unique(filters1()$year)))
   })
-  
-  
-  #update the data item choices according to the SMR selected
-  observeEvent(input$SMRaudit, {
-    updateSelectInput(session,"DataItemName", choices = c("(All)",unique(audit()$data_item_name)))
+    #Update Data Item Name selection based on inputs
+  observeEvent(to_listen_audit(), {
+    updateSelectInput(session,"DataItemName", choices = c("(All)",unique(filters1()$data_item_name)))
   })  
+    #Apply Year and Data Item Name filters to data
+  filters2 <- reactive({
+    filters1() %>%
+      filter(case_when(input$Year %in% unique(filters1()$year) ~ year == input$Year,
+                       TRUE ~ year == year),
+             case_when(input$DataItemName %in% unique(filters1()$data_item_name) ~ data_item_name == input$DataItemName,
+                       TRUE ~ data_item_name == data_item_name))
+  })
```

Comment:
Thanks for flagging this :)
 Not the most elegant solution, but I've added an if() statement so now the entry for year is only removed if the user selects year, and then selects an smr where no audit was conducted for the year selected. 
 Similarly I've added an if statement to the completeness filters, so that the data item selection is removed only in the case where it is not available for the smr selected.